Bump numpy to 2.x

xmatthias opened the pull request:
<!-- Thank you for sending your pull request. But first, have you included
unit tests, and is your code PEP8 conformant? [More details](https://github.com/freqtrade/freqtrade/blob/develop/CONTRIBUTING.md)
-->
## Summary

Update numpy to 2.x
Update ta-lib to 0.5.x

This will keep the underlying ta-lib C library at 0.4 - and will hence avoid problems while upgrading.

an update to 0.6 (the current C version) should have wheels for major platform available, otherwise it'll be a huge update pain.

## Quick changelog

- bump numpy 2
- bump ta-lib to 0.5

xmatthias committed:
chore: bump nunmpy to 2.2.4

xmatthias committed:
chore: update numpy pyproject lock range

xmatthias committed:
chore: bump ta-lib to 0.5.4

This version supports numpy2 - while still remaining on ta-lib C of 0.4
This will avoid huge update problems, as the underlying library doesn't need to be updated

xmatthias committed:
chore: bump dockerfile and windows install numpy

xmatthias committed:
chore: drop stable-baselines install from mac x64

in line with #11541

xmatthias commented:
Currently blocked by https://github.com/emergentmethods/datasieve/pull/22

xmatthias committed:
chore: update ta-lib binaries

jerome-benoit commented:
pandas-ta in version 0.3.x is also known to have issue with numpy 2.2.x.

xmatthias commented:
fully aware of this ;) 
luckily freqtrade doesn't directly / explicitly depend on pandas-ta.
While it's in requirements.txt - that's convenience for users more than anything - so one consideration is to simply drop it (though that would make all strategies depending on it automatically invalid).
It's not a future-proof dependency considering they go "paid only" anyway (read the only open issue in their repo - i'm intentionally not linking to it here).

The alternative is a "maintenance"-fork - and fixing the 5 lines that are problematic - which i did already, but didn't release just yet.

jerome-benoit commented:
What is not quite understandable is that no pandas-ta 0.4.x official releases were pushed since 0.3.x., it was tagged as such several years before the orientation change and the latest 0.4.x code is the GitHub development branch.

xmatthias committed:
chore: bump datasieve to 0.1.8

xmatthias committed:
chore: bump datasieve to 0.1.9

xmatthias committed:
chore: Mypy-exclusions to account for ta-lib import problem

xmatthias committed:
feat: update to ft-pandas-ta

Freqtrade's compatibility fork of pandas-ta
available at https://github.com/freqtrade/pandas-ta/

xmatthias committed:
chore: bump numpy to 2.2.6
